### DISCUSSION TITLE
Why I cannot use "Tech:Wind" commands of FURobot plug-ins?

### DISCUSSION THREAD

**Discussion by Shabnameh** [visose/Robots#219] [Q&A] [answered]
Hi there!! I am working on the robotic fabrication of carbon fibre for my master final thesis. everything about the definition of the robot and the code generation is fine, I have ran everything. But at this point I need to optimize the toolpath, or let's say the trajectory which end-effector will pass to wind my intended pattern but I don't know what exactly to do. During my search I found the "Tech:Wind" commands of the FURobot plug-ins in which there seem to be some tools to optimize the winding path, but I cannot use the tools. First I should say that I see the sign/emoji of the lock beside them all, then once I try to use and run them, it gives me a warning as: "This component is for internal test". What does it mean? I cannot use them? It needs any licence or so on?

I would br wondering if anyone had any idea about how I can define the toolpath easier for the robot to reach. I should explain more in this way that, the pattern is composed of a series of Polylines drawn among each 2 points of a series of points around a rectangle frame, as a drawing they all are straight lines. But as a winding patterns, in reality, they should be rounded around the pins at the location of the points, and to do that, the robot needs to also have a zone of freedom to round around the pins and so. for this I need to optimize/transform the linear pattern to a more adaptive trajectory which the robot's head, the end effector, will pass through. And I need some help/idea how to implement this. 

Thank you in advanced.

**Comment by visose** [comment, marked as answer]
Those `Tech:Wind` components belong to the FURobot plugin, not Robots. A lock/internal-test warning means they are not exposed as normal public components from that plugin. For Robots, generate the winding/path geometry in Grasshopper or another tool, then feed the resulting target planes/toolpath into Robots for simulation and code generation.
